### PR TITLE
Feat human vs human

### PR DESCRIPTION
**Pull request by parkerpersaud**
https://trello.com/c/fzZyu7N6/48-600-human-vs-human-game

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -3,7 +3,7 @@
   :url "https://github.com/klkelley/tictactoe-clojure"
   :license {:name "Eclipse Public License"
             :url "http://www.eclipse.org/legal/epl-v10.html"}
-  :dependencies [[org.clojure/clojure "1.9.0"]]
+  :dependencies [[org.clojure/clojure "1.9.0"] [org.clojure/math.numeric-tower "0.0.4"]]
```
Comment (aaronlahey) [inline project.clj:6]: woah.

File: src/tictactoe/game_runner.clj [hunk 2]
```diff
@@ -1,35 +1,18 @@
 (ns tictactoe.game-runner
-  (:require [clojure.string :as string]))
+  (:require [clojure.string :as string]
+            [clojure.math.numeric-tower :as math]))
 
-(def game-types
-  {"1" "Human vs. Human" "2" "Human vs. Easy Computer"})
+(def ^:dynamic sleep-time 5000)
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: The convention for dynamic vars is to surround its name in earmuffs: `*sleep-time*`

File: test/tictactoe/game_runner_test.clj [hunk 3]
```diff
@@ -1,73 +1,88 @@
...
-                  "game => Human vs. Human"
-                  "board => 4x4"
-                  "player1 => A"
-                  "player2 => B\n")
-           (with-out-str ""
-             (with-in-str (make-input '("1" "2" "A" "a" "B")) (game-runner/start)))))))
\ No newline at end of file
+(deftest test-cell-choice-validation
+  (testing "User will continue to be prompted for valid input and available spaces"
+  (is (= true (string/includes?
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 4]
```diff
@@ -1,73 +1,88 @@
...
+  (testing "Game can be won via right diagonal"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "1" "X" "O" "2" "3" "4" "7" "6")) 
+            (binding [sleep-time 0]
+              (start)))) "Player X wins!" )))))
 
+(deftest win-game-via-column
+  (testing "Game can be won via column"
+  (is (= true (string/includes?
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation.

File: test/tictactoe/game_runner_test.clj [hunk 5]
```diff
@@ -1,73 +1,88 @@
...
          (with-out-str ""
-           (with-in-str (make-input '("4" "1" "2" "a" "B")) (game-runner/start)))))))
+           (with-in-str (make-input '("1" "1" "X" "O" "0" "2" "4" "1" "8")) 
+            (binding [sleep-time 0]
+              (start)))) "Player X wins!" )))))
 
+(deftest win-game-diagonally-right
+  (testing "Game can be won via right diagonal"
+  (is (= true (string/includes?
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation.

File: test/tictactoe/game_runner_test.clj [hunk 6]
```diff
@@ -1,73 +1,88 @@
...
-                "Please pick player 1's marker (A-Z)"
-                "Please pick player 2's marker (A-Z)"
-                "You've chosen:"
-                "game => Human vs. Human"
-                "board => 4x4"
-                "player1 => A"
-                "player2 => B\n")
+(deftest win-game-diagonally
+  (testing "Game can be won via left diagonal"
+  (is (= true (string/includes?
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 7]
```diff
@@ -1,73 +1,88 @@
...
+  (testing "User must choose an available menu option"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("3" "1" "1" "x" "p" "0" "4" "1" "3" "2"))
+            (binding [sleep-time 0]
+              (start)))) "Invalid. Please pick an available option." )))))
 
+(deftest play-tie-game
+  (testing "Game can be played to a tie"
+  (is (= true (string/includes?
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 8]
```diff
@@ -1,73 +1,88 @@
...
-                  "You've chosen:"
-                  "game => Human vs. Human"
-                  "board => 4x4"
-                  "player1 => A"
-                  "player2 => B\n")
-           (with-out-str ""
-             (with-in-str (make-input '("1" "2" "a" "b")) (game-runner/start)))))))
+(deftest invalid-menu-choice
+  (testing "User must choose an available menu option"
+  (is (= true (string/includes?
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 9]
```diff
@@ -1,73 +1,88 @@
...
+  (testing "Player one wins the game on a 4x4 board"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "2" "X" "O" "0" "4" "1" "8" "2" "13" "3")) 
+            (binding [sleep-time 0]
+              (start)))) "Player X wins!" )))))
+
+(deftest invalid-marker-choice
+  (testing "User will be prompted for different marker if choice is invalid"
+  (is (= true (string/includes?
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 10]
```diff
@@ -1,73 +1,88 @@
...
+  (testing "Player two wins the game on a 3x3 board"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "1" "X" "O" "1" "4" "2" "8" "7" "0")) 
+           (binding [sleep-time 0]
+            (start)))) "Player O wins!" )))))
+
+(deftest test-4x4-game-human-v-human-player1-wins
+  (testing "Player one wins the game on a 4x4 board"
+  (is (= true (string/includes?
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 11]
```diff
@@ -1,73 +1,88 @@
...
+  (testing "Player one wins the game on a 3x3 board via row"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "1" "X" "O" "0" "5" "1" "4" "2")) 
+            (binding [sleep-time 0]
+              (start)))) "Player X wins!" )))))
+
+(deftest test-game-human-v-human-player2-wins
+  (testing "Player two wins the game on a 3x3 board"
+  (is (= true (string/includes?
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 12]
```diff
@@ -1,73 +1,88 @@
...
 (defn make-input
   [coll]
   (apply str (interleave coll (repeat "\n"))))
 
-(defn lines
-  [& args]
-  (apply str (interpose "\n" args)))
+(deftest test-game-play-human-v-human
+  (testing "Player one wins the game on a 3x3 board via row"
+  (is (= true (string/includes?
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: src/tictactoe/game_runner.clj [hunk 13]
```diff
@@ -121,24 +104,164 @@
...
+(defn ^:private winner-exists?
+  [board combo]
+  (if (has-three-in-a-row? board combo)
+    (board (first combo))
+    nil))
+
+(defn ^:private winner? [board]
+  (some #(winner-exists? board %) (winning-combinations board)))
+
+(defn ^:private game-results [board]
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: arguments should be on the first or second line consistently throughout.

File: src/tictactoe/game_runner.clj [hunk 14]
```diff
@@ -121,24 +104,164 @@
...
+
+(defn ^:private right-diagonal
+  [board]
+  (let [right-corner (dec (grid-size board))]
+  (reduce #(conj %1 (+ right-corner (* %2 right-corner)))
+          [right-corner] (range 1 (grid-size board)))))
+
+(defn ^:private diagonals
+  [board]
+    [(left-diagonal board)
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: indentation

File: src/tictactoe/game_runner.clj [hunk 15]
```diff
@@ -121,24 +104,164 @@
...
+  [choices]
+  [{:type :human :marker (choices :player1)} {:type :human :marker (choices :player2)}])
+
+(defn ^:private make-board 
+  [board-choice]
+  (if (= "3x3" board-choice)
+    (vec (take 9 (range)))
+    (vec (take 16 (range)))))
+
+(defn ^:private empty-space? [space]
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: arguments on the wrong line

File: src/tictactoe/game_runner.clj [hunk 16]
```diff
@@ -121,24 +104,164 @@
...
+  (number? space))
+
+(defn ^:private available-spaces
+  [board]
+  (filter #(empty-space? %) board))
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: You don't need an anonymous fn here.

File: src/tictactoe/game_runner.clj [hunk 17]
```diff
@@ -121,24 +104,164 @@
...
+  (math/sqrt (count board)))
+
+(defn ^:private clear-screen
+  []
+  (print (str (char 27) "[2J"))
+  (print (str (char 27) "[;H")))
+
+(defn ^:private display-board 
+  [board]
+  (println (apply str (map #(str "  "(board %)"  " (board-dividers % (grid-size board)))
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: Spacing here. Also, this is pretty hard to visually parse. I'd probably use a `let` with some self-documenting variable names.

File: src/tictactoe/game_runner.clj [hunk 18]
```diff
@@ -121,24 +104,164 @@
...
+  [board combo]
+  (= 1 (count (set (map #(board %) combo)))))
+
+(defn ^:private winner-exists?
+  [board combo]
+  (if (has-three-in-a-row? board combo)
+    (board (first combo))
+    nil))
+
+(defn ^:private winner? [board]
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: arguments on the wrong line

File: src/tictactoe/game_runner.clj [hunk 19]
```diff
@@ -121,24 +104,164 @@
...
+  (some #(winner-exists? board %) (winning-combinations board)))
+
+(defn ^:private game-results [board]
+  (let [winner (winner? board)]
+    (if (nil? winner)
+      (println "\nIts a tie!")
+      (println "\nPlayer" winner "wins!"))))
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: I find prepending or appending a `"\n"` to random strings to be confusing. It kind of couples these low level functions which know how to print a message to the higher-level functions which are orchestrating all of these `print`s into a single view. If that doesn't make sense I can explain further.

File: test/tictactoe/game_runner_test.clj [hunk 20]
```diff
@@ -1,73 +1,88 @@
...
+         (with-out-str ""
+           (with-in-str (make-input '("1" "1" "X" "O" "0" "5" "1" "4" "2")) 
+            (binding [sleep-time 0]
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 21]
```diff
@@ -1,73 +1,88 @@
...
+              (start)))) "Player X wins!" )))))
+
+(deftest test-game-human-v-human-player2-wins
+  (testing "Player two wins the game on a 3x3 board"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "1" "X" "O" "1" "4" "2" "8" "7" "0")) 
+           (binding [sleep-time 0]
+            (start)))) "Player O wins!" )))))
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 22]
```diff
@@ -1,73 +1,88 @@
...
+
+(deftest test-4x4-game-human-v-human-player1-wins
+  (testing "Player one wins the game on a 4x4 board"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "2" "X" "O" "0" "4" "1" "8" "2" "13" "3")) 
+            (binding [sleep-time 0]
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 23]
```diff
@@ -1,73 +1,88 @@
...
+              (start)))) "Player X wins!" )))))
+
+(deftest invalid-marker-choice
+  (testing "User will be prompted for different marker if choice is invalid"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "1" "X" "5" "P" "0" "4" "1" "3" "2")) 
+            (binding [sleep-time 0]
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 24]
```diff
@@ -1,73 +1,88 @@
...
-                  "player1 => A"
-                  "player2 => B\n")
-           (with-out-str ""
-             (with-in-str (make-input '("1" "2" "a" "b")) (game-runner/start)))))))
+(deftest invalid-menu-choice
+  (testing "User must choose an available menu option"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("3" "1" "1" "x" "p" "0" "4" "1" "3" "2"))
+            (binding [sleep-time 0]
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 25]
```diff
@@ -1,73 +1,88 @@
...
+              (start)))) "Invalid. Please pick an available option." )))))
 
+(deftest play-tie-game
+  (testing "Game can be played to a tie"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "1" "x" "p" "0" "4" "1" "2" "6" "3" "5" "7" "8")) 
+            (binding [sleep-time 0]
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 26]
```diff
@@ -1,73 +1,88 @@
...
-                "board => 4x4"
-                "player1 => A"
-                "player2 => B\n")
+(deftest win-game-diagonally
+  (testing "Game can be won via left diagonal"
+  (is (= true (string/includes?
          (with-out-str ""
-           (with-in-str (make-input '("4" "1" "2" "a" "B")) (game-runner/start)))))))
+           (with-in-str (make-input '("1" "1" "X" "O" "0" "2" "4" "1" "8")) 
+            (binding [sleep-time 0]
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 27]
```diff
@@ -1,73 +1,88 @@
...
+              (start)))) "Player X wins!" )))))
 
+(deftest win-game-diagonally-right
+  (testing "Game can be won via right diagonal"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "1" "X" "O" "2" "3" "4" "7" "6")) 
+            (binding [sleep-time 0]
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 28]
```diff
@@ -1,73 +1,88 @@
...
+              (start)))) "Player X wins!" )))))
 
+(deftest win-game-via-column
+  (testing "Game can be won via column"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "1" "X" "O" "2" "3" "5" "7" "8")) 
+            (binding [sleep-time 0]
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 29]
```diff
@@ -1,73 +1,88 @@
...
-                  "player2 => B\n")
-           (with-out-str ""
-             (with-in-str (make-input '("1" "2" "A" "a" "B")) (game-runner/start)))))))
\ No newline at end of file
+(deftest test-cell-choice-validation
+  (testing "User will continue to be prompted for valid input and available spaces"
+  (is (= true (string/includes?
+         (with-out-str ""
+           (with-in-str (make-input '("1" "1" "x" "p" "0" "0" "f" "4" "1" "2" "6" "3" "5" "7" "8")) 
+            (binding [sleep-time 0]
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: test/tictactoe/game_runner_test.clj [hunk 30]
```diff
@@ -1,73 +1,88 @@
...
+            (start)))) "Its a tie!" )))))
```
Comment (aaronlahey) [inline test/tictactoe/game_runner_test.clj]: indentation

File: src/tictactoe/game_runner.clj [hunk 31]
```diff
@@ -1,35 +1,18 @@
...
 
-(def board-sizes
+(def ^:private game-types
+  {"1" "Human vs. Human" "2" "Human vs. Impossible Computer"})
+
+(def ^:private board-sizes
   {"1" "3x3" "2" "4x4"})
 
-(def markers 
+(def ^:private markers
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj:13]: I was able to change the values in this set and your tests still passed

File: src/tictactoe/game_runner.clj [hunk 32]
```diff
@@ -121,24 +104,164 @@
...
+    (board (first combo))
+    nil))
+
+(defn ^:private winner? [board]
+  (some #(winner-exists? board %) (winning-combinations board)))
+
+(defn ^:private game-results [board]
+  (let [winner (winner? board)]
+    (if (nil? winner)
+      (println "\nIts a tie!")
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: I was able to remove these newlines and the tests still passed

File: src/tictactoe/game_runner.clj [hunk 33]
```diff
@@ -121,24 +104,164 @@
...
+  [board combo]
+  (if (has-three-in-a-row? board combo)
+    (board (first combo))
+    nil))
+
+(defn ^:private winner? [board]
+  (some #(winner-exists? board %) (winning-combinations board)))
+
+(defn ^:private game-results [board]
+  (let [winner (winner? board)]
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: Generally functions ending in `?` suggest a predicate returning `true` or `false`. I would suggest renaming this function to `winner` and `winner?` being implemented as `(some? (winner board))`

File: src/tictactoe/game_runner.clj [hunk 34]
```diff
@@ -121,24 +104,164 @@
...
+  (print (str (char 27) "[;H")))
+
+(defn ^:private display-board 
+  [board]
+  (println (apply str (map #(str "  "(board %)"  " (board-dividers % (grid-size board)))
+                            (range (count board))))))
+
+(defn ^:private rows 
+  [board]
+  (partition (grid-size board) (range (count board))))
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: Your tests passed when I hard-coded this to...

```
(defn ^:private rows
  [_]
  [[0 1 2]
   [3 4 5]])
```

File: src/tictactoe/game_runner.clj [hunk 35]
```diff
@@ -121,24 +104,164 @@
...
+
+(defn ^:private columns 
+  [board]
+  (apply map vector (rows board)))
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj:213]: Your tests passed when I harded-coded this to...

```
(defn ^:private columns
  [board]
  [[2 5 8]])
```

File: src/tictactoe/game_runner.clj [hunk 36]
```diff
@@ -121,24 +104,164 @@
...
+(defn ^:private grid-size
+  [board]
+  (math/sqrt (count board)))
+
+(defn ^:private clear-screen
+  []
+  (print (str (char 27) "[2J"))
+  (print (str (char 27) "[;H")))
+
+(defn ^:private display-board
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: This implementation passes your tests

```
(defn ^:private display-board
  [board]
  "")
```

File: src/tictactoe/game_runner.clj [hunk 37]
```diff
@@ -121,24 +104,164 @@
...
+    (get-human-move (available-spaces board))))
+
+(defn ^:private row-divider
+  [length]
+  (->> 
+    "--"
+    (repeat (* length length))
+    (apply str)))
+
+(defn ^:private board-dividers
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: This implementation passes your tests

```
(defn ^:private board-dividers
  [number grid-size]
  "")
```

File: src/tictactoe/game_runner.clj [hunk 38]
```diff
@@ -121,24 +104,164 @@
...
+    (if (and choice (some #(= choice %) available-spaces))
+      choice
+      (recur available-spaces))))
+
+(defn ^:private next-move
+  [current-player board]
+  (if (= (current-player :type) :human)
+    (get-human-move (available-spaces board))))
+
+(defn ^:private row-divider
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: This implementation passes your tests

```
(defn ^:private row-divider
  [length]
  "")
```

File: src/tictactoe/game_runner.clj [hunk 39]
```diff
@@ -121,24 +104,164 @@
...
+    (Integer. (re-find #"^\d+$" choice))
+    (catch Exception e nil)))
+
+(defn ^:private place-move 
+  [spot marker board]
+  (assoc board spot marker))
+
+(defn ^:private get-human-move
+  [available-spaces]
+  (println "\nPlease pick an available cell: ")
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: This line isn't tested.

File: src/tictactoe/game_runner.clj [hunk 40]
```diff
@@ -121,24 +104,164 @@
...
+
+(defn ^:private available-spaces
+  [board]
+  (filter #(empty-space? %) board))
+
+(defn ^:private to-integer 
+  [choice]
+  (try
+    (Integer. (re-find #"^\d+$" choice))
+    (catch Exception e nil)))
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: I changed this to `(catch Exception e 0)` and your tests passed

File: src/tictactoe/game_runner.clj [hunk 41]
```diff
@@ -121,24 +104,164 @@
...
-  (loop-and-print game-options))
+  (print-message "You've chosen: ")
+  (display-game-menu-options game-options))
 
 (defn ^:private menu
   [game-types board-sizes]
-  (print-message (welcome-message))
-  (let [game-choice (pick game-types (game-type-message))
-        board-size (pick board-sizes (board-type-message))
+  (print-message "Welcome to Tic Tac Toe!\n")
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj]: This line isn't tested

File: src/tictactoe/game_runner.clj [hunk 42]
```diff
@@ -1,35 +1,18 @@
...
-
-(defn ^:private player-marker-message
-  [player]
-  (str "Please pick " player "'s marker (A-Z)"))
-
-(defn ^:private game-choices-message
-  []
-  "You've chosen:")
-
 (defn ^:private print-message
```
Comment (aaronlahey) [inline src/tictactoe/game_runner.clj:16]: Your tests pass when I comment out the second `[message supplementing-info]` version of this function

**Review by aaronlahey — CHANGES_REQUESTED** [review]
A few notes:

1. In a lot of cases your tests never fail, they just hang forever. That's not ideal.

2. There are a lot of indentation issues

3. There are a lot of holes in your test coverage

4. At ~250 lines it's probably about time to split out a new namespace or two. Once this feedback is addressed I'd be open to a refactor PR.

File: src/tictactoe/game_runner.clj [hunk 43]
```diff
@@ -121,24 +104,164 @@
...
+  [available-spaces]
+  (println "\nPlease pick an available cell: ")
+  (let [choice (to-integer (get-input))]
+    (if (and choice (some #(= choice %) available-spaces))
+      choice
+      (recur available-spaces))))
+
+(defn ^:private next-move
+  [current-player board]
+  (if (= (current-player :type) :human)
```
Comment (andarcabrera) [inline src/tictactoe/game_runner.clj]: Is it a little early to check if the player is human? It seems at this point the game can default to human move

File: src/tictactoe/game_runner.clj (hunk 17 above)
Comment (andarcabrera) [reply src/tictactoe/game_runner.clj]: The 4x4 board needs some prettifying.

File: src/tictactoe/game_runner.clj (hunk 42 above)
Comment (parkerpersaud) [reply src/tictactoe/game_runner.clj:16]: That's interesting - when I comment it out all my tests fail due to wrong number of arguments.

File: project.clj (hunk 1 above)
Comment (parkerpersaud) [reply project.clj:6]: woah in a bad way or woah in a good way?

Comment (aaronlahey) [reply project.clj:6]: I've just never seen a project depend on this library. You basically just pulled out a chainsaw to slice up an apple. :) Clojure libs tend to be really stable so there's nothing wrong with taking on this dependency.

File: src/tictactoe/game_runner.clj (hunk 19 above)
Comment (parkerpersaud) [reply src/tictactoe/game_runner.clj]: So would you recommend omitting the newlines?

File: test/tictactoe/game_runner_test.clj (hunk 20 above)
Comment (parkerpersaud) [reply test/tictactoe/game_runner_test.clj]: I will fix these - I formatted them using vim and was trying to keep things consistent.

File: src/tictactoe/game_runner.clj (hunk 19 above)
Comment (aaronlahey) [reply src/tictactoe/game_runner.clj]: Yep. Let some higher level function put the newlines where necessary.

File: test/tictactoe/game_runner_test.clj (hunk 20 above)
Comment (aaronlahey) [reply test/tictactoe/game_runner_test.clj]: 👍

File: src/tictactoe/game_runner.clj (hunk 42 above)
Comment (aaronlahey) [reply src/tictactoe/game_runner.clj:16]: Sorry. Poorly worded. I just commented out the implementation of that arity.

```
(defn ^:private print-message
  ([message]
   (println message))
  ([message supplementing-info]
   #_(println message "=>" supplementing-info)))
```

**Review by aaronlahey — APPROVED** [review]

**Review by andarcabrera — APPROVED** [review]

**Pull request merged**